sdk/java-sdk-spring/src/main/java/kalix/javasdk/client/ComponentClient.java

```diff
@@ -40,10 +42,20 @@ public class ComponentClient {
 
   private final KalixClient kalixClient;
 
+  private Optional<Metadata> callMetadata = Optional.empty();
+
   public ComponentClient(KalixClient kalixClient) {
     this.kalixClient = kalixClient;
   }
 
+  public void setCallMetadata(Metadata callMetadata) {
```

Comment:
Should we document that this is intended for internal use? or is it ok for users to use this? if so, having this and the `.withMetadata` after building the deferred call might be confusing...

Reply:
Right. Actually this is a oversight of mine. I was planning to make it internal API. 

On the other hand, now I'm wondering what is the impact for using `withMetadata` in the `DeferredCall`. In both impls, we are setting the metadata, but we probably want to copy the tracing over, if any.